developer/openshift/dev_setup.sh

```diff
@@ -229,26 +236,32 @@ install_pipeline_service() {
   "$PROJECT_DIR/operator/images/access-setup/content/bin/setup_work_dir.sh" \
     ${DEBUG:+"$DEBUG"} \
     --work-dir "$WORK_DIR" \
-    --kustomization "git::$GIT_URL/developer/openshift/gitops/argocd?ref=$GIT_REF" |
+    --kustomization "$GIT_URL/developer/openshift/gitops/argocd?ref=$GIT_REF" |
     indent 2
 
+  # Patch the url/branch to target the expected repository/branch
+  manifest_dir="$(find "$WORK_DIR/environment/compute" -mindepth 1 -maxdepth 1 -type d)"
+  for app in "pipeline-service" "pipeline-service-storage"; do
+    cat << EOF >"$manifest_dir/patch-$app.yaml"
+---
+apiVersion: argoproj.io/v1alpha1
+kind: Application
+metadata:
+  name: $app
+  namespace: openshift-gitops
+spec:
+  source:
+    repoURL: $GIT_URL
+    targetRevision: $GIT_REF
+EOF
+    yq -i ".patches += [\"patch-$app.yaml\"]" "$manifest_dir/kustomization.yaml"
```

Comment:
Because of kustomize #[4911](https://github.com/kubernetes-sigs/kustomize/pull/4911), it has to use patchesStrategicMerge explicitly. 
`yq -i ".patchesStrategicMerge += [\"patch-$app.yaml\"]" "$manifest_dir/kustomization.yaml"`

Reply:
we can use the way [Patch using Path Strategic Merge](https://kubectl.docs.kubernetes.io/references/kustomize/kustomization/patches/#patch-using-path-strategic-merge)
`yq -i ".patches += [{\"path\": \"patch-$app.yaml\"}]"  "$manifest_dir/kustomization.yaml"`

Reply:
Fixed. You can merge if there are no other comments.